src/Servers/Kestrel/Core/src/ListenOptions.cs

```diff
@@ -75,7 +75,7 @@ internal ListenOptions(ulong fileHandle, FileHandleType handleType)
         /// Enables connection middleware to resolve and use services registered by the application during startup.
         /// Only set if accessed from the callback of a <see cref="KestrelServerOptions"/> Listen* method.
         /// </summary>
-        public KestrelServerOptions KestrelServerOptions { get; internal set; }
+        public KestrelServerOptions KestrelServerOptions { get; internal set; } = default!; // TODO - Always available?
```

Comment:
Given that KestrelServerOptions wasn't manually constructed and this was instead accessed via a ConfigureKestrel/UseKestrel configureOptions callback, then yes. If someone just calls `new KestrelServerOptions()`, then no.

I think you theoretically could do something like `(new KestrelServerOptions()).ListenLocalhost(5000, listenOptions => { ... });` and get a ListenOptions object without KestrelServerOption set, but I think this is just like KestrelServerOptions.ApplicationServices where we can expect this to be set the way it's typically used and I agree we should annotate it as non-nullable.